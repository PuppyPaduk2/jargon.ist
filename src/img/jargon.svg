<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 463 446"><defs><filter id="a" width="103.1%" height="103.2%" x="-1.5%" y="-1.1%" filterUnits="objectBoundingBox"><feOffset dy="2" in="SourceAlpha" result="shadowOffsetOuter1"/><feGaussianBlur in="shadowOffsetOuter1" result="shadowBlurOuter1" stdDeviation="2"/><feColorMatrix in="shadowBlurOuter1" result="shadowMatrixOuter1" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.1 0"/><feMerge><feMergeNode in="shadowMatrixOuter1"/><feMergeNode in="SourceGraphic"/></feMerge></filter></defs><g fill="none" fill-rule="nonzero" filter="url(#a)" transform="translate(4 2)"><path fill="#4C5BE6" d="M0 0h202v12H0zM0 112h202v12H0z"/><path fill="#717A85" d="M0 28h303v12H0zM0 172h430v12H0zM0 282h400v12H0zM0 228h445v12H0zM0 200h121v12H0zM0 256h60v12H0zM186 256h80v12h-80zM392 256h63v12h-63zM239 200h216v12H239z"/><path fill="#FD5A57" d="M129 200h101v12H129z"/><path fill="#4C5BE6" d="M68 256h110v12H68z"/><path fill="#E6964C" d="M274 256h110v12H274z"/><path fill="#717A85" d="M0 56h270v12H0zM0 84h303v12H0zM0 342h270v12H0z"/><path fill="#FD5A57" d="M0 370h180v12H0z"/><path fill="#717A85" d="M0 398h210v12H0zM0 426h260v12H0z"/></g></svg>