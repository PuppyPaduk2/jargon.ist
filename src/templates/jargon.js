// @flow
/* eslint-disable react/no-danger */

import React, { Fragment } from 'react';
import Helmet from 'react-helmet';
import Link from 'gatsby-link';
import { graphql } from 'graphql';

type Props = {
  data: Object,
};

const Jargon = ({ data }: Props) => {
  const { markdownRemark } = data;
  const { frontmatter, html, fields } = markdownRemark;
  return (
    <div>
      <Helmet>
        <title>{frontmatter.title}</title>
        <meta name="keywords" content={frontmatter.tags.join(', ')} />
      </Helmet>

      <h1>{frontmatter.title}</h1>
      <div dangerouslySetInnerHTML={{ __html: html }} />
      <hr />
      <div>
        <strong>Etiketler:</strong>
        {fields.tagList.map((tag, index) => (
          <Fragment key={tag.slug}>
            <Link to={`/e/${tag.slug}`}>{tag.title}</Link>
            {index < fields.tagList.length - 1 && <span>, </span>}
          </Fragment>
        ))}
      </div>
    </div>
  );
};

export const pageQuery = graphql`
  query JargonBySlug($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      html
      fields {
        tagList {
          title
          slug
        }
      }
      frontmatter {
        title
        tags
      }
    }
  }
`;

export default Jargon;

/* eslint-enable react/no-danger */
