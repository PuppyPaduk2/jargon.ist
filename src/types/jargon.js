// @flow

import { type Tag } from './tag';

export type JargonFrontmatter = {
  title: string,
  sameWith?: Array<string>,
  tags?: Array<string>,
};

export type Jargon = {
  html: string,
  title: string,
  slug: string,
  tags?: Array<Tag>,
  sameWith?: Array<string>,
};
