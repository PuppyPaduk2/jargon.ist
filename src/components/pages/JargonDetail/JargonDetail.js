// @flow
/* eslint-disable react/no-danger */

import React, { Fragment } from 'react';
import Link from 'gatsby-link';

import { type Jargon } from '../../../types/jargon';

type Props = {
  jargon: Jargon,
};

const JargonDetail = ({ jargon }: Props) => (
  <Fragment>
    <h1 className="u-text-lowercase">{jargon.title}</h1>
    <div dangerouslySetInnerHTML={{ __html: jargon.html }} />

    {(jargon.tags || []).length > 0 && (
      <Fragment>
        <hr className="u-gap-top-medium" />

        <div>
          {(jargon.tags || []).map(tag => (
            <Link
              key={tag.slug}
              to={`/k/${tag.slug}`}
              className="u-gap-right-small u-text-lowercase"
            >
              {tag.title}
            </Link>
          ))}
        </div>
      </Fragment>
    )}
  </Fragment>
);

export default JargonDetail;

/* eslint-enable react/no-danger */
