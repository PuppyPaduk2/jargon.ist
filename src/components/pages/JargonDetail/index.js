// @flow

import JargonDetail from './JargonDetail';

export default JargonDetail;
