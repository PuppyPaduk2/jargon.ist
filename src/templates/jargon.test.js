// @flow

import React from 'react';
import { shallow } from 'enzyme';

import { Jargon } from './jargon';

import JargonDetail from '../components/pages/JargonDetail';

const data = {
  markdownRemark: {
    html: '<p>hello world</p>',
    frontmatter: {
      title: 'hello',
    },
    fields: {
      tagList: [],
    },
  },
};

it('renders correctly', () => {
  const wrapper = shallow(<Jargon data={data} />);

  expect(wrapper).toMatchSnapshot();
});

it('renders title', () => {
  const wrapper = shallow(<Jargon data={data} />);

  expect(wrapper.find('title')).toHaveLength(1);
});

it('renders JargonDetail component', () => {
  const wrapper = shallow(<Jargon data={data} />);

  expect(wrapper.find(JargonDetail)).toHaveLength(1);
});
