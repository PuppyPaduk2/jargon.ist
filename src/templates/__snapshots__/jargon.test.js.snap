// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      hello
    </title>
    <meta
      content=""
      name="keywords"
    />
    <meta
      content="hello"
      name="og:title"
    />
    <meta
      content="hello"
      name="twitter:title"
    />
  </HelmetWrapper>
  <JargonDetail
    jargon={
      Object {
        "html": "<p>hello world</p>",
        "slug": "hello",
        "tags": Array [],
        "title": "hello",
      }
    }
  />
</React.Fragment>
`;
